src/sql/connection.py

```diff
@@ -348,3 +353,13 @@ def _get_curr_connection_info(self):
             "driver": getattr(engine.dialect, "driver", None),
             "server_version_info": getattr(engine.dialect, "server_version_info", None),
         }
+
+    def _transpile_query(self, query):
+        connection_info = self._get_curr_connection_info()
+        try:
+            write_dialect = DIALECT_NAME_SQLALCHEMY_TO_SQLGLOT_MAPPING.get(
+                connection_info["dialect"], connection_info["dialect"]
+            )
+            query = sqlglot.transpile(query, read="duckdb", write=write_dialect)[0]
+        finally:
+            return query
```

Comment:
why do we need this try ... finally?

Reply:
Since `sqlglot` might throw the Exception when the write_dialect name isn't supported, e.g. `sqlglot.transpile(query, read="duckdb", write="postgresql")[0]`, I am thinking when the write_dialect isn't support by the `sqlglot`, we can still safely return the original query

Reply:
ah. makes sense. let's add some telemetry when transpile fails